feedback/migrations/0020_remove_feedbacktag_feedbacks_conversation_and_more.py

```diff
@@ -7,26 +7,30 @@
 def generate_conversations_for_old_feedback(apps, schema_editor):
     Conversation = apps.get_model("feedback", "Conversation")
     Feedback = apps.get_model("feedback", "Feedback")
-    for fb in Feedback.objects.only('exercise', 'student', 'conversation'):
-        conv, _created = Conversation.objects.get_or_create(
-            exercise=fb.exercise,
-            student=fb.student,
-        )
+    fbs = list(Feedback.objects.only("exercise", "student", "conversation"))
+    new_convs = {}
+    for fb in fbs:
+        conv = new_convs.get((fb.exercise, fb.student))
```

Comment:
Since the dictionary key is not needed for saving the objects in the database, I was thinking of simplifying the key. The key tuple uses the objects now, which should be fine when the hash value is well-defined for the object. I don't know now how Django objects define the hash, but if the key used the ids directly, then the key would only consist of basic built-in Python types.
```suggestion
        conv = new_convs.get((fb.exercise.pk, fb.student.pk))
```

Reply:
done